File: frontend/src/types/translation.ts
export type TranslationTask = {
  task_id: string
  status: string
  target_language: string
  result_file_id: string | null
  result_filename: string | null
  error: string | null
  created_at: string
  updated_at: string
}

export type CreateTranslationTaskPayload = {
  file_id: string
  target_language: string
  api_key?: string
}

File: backend/app/schemas/translation.py
from datetime import datetime

from pydantic import BaseModel, Field


class TranslationTaskCreate(BaseModel):
    file_id: str
    target_language: str = Field(min_length=1)
    api_key: str | None = None


class TranslationTaskRead(BaseModel):
    task_id: str
    status: str
    target_language: str
    result_file_id: str | None = None
    result_filename: str | None = None
    error: str | None = None
    created_at: datetime
    updated_at: datetime
